src/vmm/src/guest_config/x86/mod.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::guest_config::Modifier;
+
+/// Composite type that holistically provides
+/// the location of a specific register being used
+/// in the context of a CPUID tree
+#[allow(unused)]
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct CpuidRegisterId {
+    leaf: u32,
+    subleaf: u32,
+    pointer: CpuidRegister,
+}
+
+/// CPUID register enumeration
+#[allow(unused)]
+#[allow(missing_docs)]
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub enum CpuidRegister {
+    Eax,
+    Ebx,
+    Ecx,
+    Edx,
+}
+
+/// Template providing bitmask modifier info
+/// for CPUID and MSRs.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct X86CpuTemplate {
+    cpuid_modifiers: HashMap<CpuidRegisterId, Modifier>,
```

Comment:
Are we not going to suffer from issues similar to https://github.com/firecracker-microvm/firecracker/pull/3445 when using HashMap?

Reply:
> Are we not going to suffer from issues similar to #3445 when using HashMap?

Not quite. The reproducibility is related to ordering of entries in the map which we don't want or need. Although thinking about it a bit more, using a Set here is more desirable. There shouldn't be a need to look up the entries, just iterate through them. I will need to change this.

Reply:
Why is a set better than a vector here?

Reply:
This is out of date, but In case this becomes relevant again, instead of using a map. The HashSet enforces that there are no duplicates in the collection. Not a big deal though, this will be validated against, but it is my preference to have defensive structures in place.